// electron-logger/lib/settings.js
// Generated by CoffeeScript 1.9.2
(function() {
  var ENV, colorsArray;

  ENV = {
    dev: "development",
    pro: "production"
  };

  colorsArray = {
    BLACK: "black",
    RED: "red",
    GREEN: "green",
    YELLOW: "yellow",
    BLUE: "blue",
    MAGENTA: "magenta",
    CYAN: "cyan",
    WHITE: "white",
    GRAY: "gray",
    GREY: "grey"
  };

  module.exports = {
    LEVELS: {
      ERROR: 0,
      INFO: 1,
      DEBUG: 2
    },
    COLORS: [colorsArray.RED, colorsArray.GREEN, colorsArray.BLUE],
    STATUS: ["closed", "open"],
    environment: ENV.dev,
    dest: {
      terminal: true,
      file: false
    },
    level: "INFO",
    setLevelByString: function(levelStr) {
      return this.level = levelStr.toUpperCase();
    },
    setLevelByNumber: function(levelNum) {
      var k, ref, v;
      ref = this.LEVELS;
      for (k in ref) {
        v = ref[k];
        if (v === levelNum) {
          this.setLevelByString(k);
          return;
        }
      }
    }
  };

}).call(this);
